core/src/main/scala/algebra/lattice/Bool.scala

```diff
@@ -28,17 +28,17 @@ trait Bool[@sp(Int, Long) A] extends Any with Heyting[A] { self =>
 
   override def dual: Bool[A] = new DualBool(this)
   /**
-   * Every Boolean algebra is a CommutativeRing, but we don't extend
-   * CommutativeRing because, e.g. we might want a Bool[Int] and CommutativeRing[Int] to
+   * Every Boolean algebra is a BoolRing, but we don't extend BoolRing
+   * because, e.g. we might want a Bool[Int] and CommutativeRing[Int] to
    * refer to different structures, by default.
    */
-  def asCommutativeRing: CommutativeRing[A] =
-    new CommutativeRing[A] {
+  def asCommutativeRing: BoolRing[A] =
```

Comment:
Should we change the name perhaps to `asBoolRing`? It is easy to mistype `Ring` when instead of `Rig` or vice-versa.

It's a bit weird that if you just need a `Rig` there are two ways to get it: `asCommutativeRig` and `asCommutativeRing` on a Bool. We added the comment in the README.md about this. Maybe that documentation should also appear here.